domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/config/beanpostprocessor/NatsControllerBeanPostProcessor.kt

```diff
@@ -0,0 +1,36 @@
+package systems.ajax.malov.stockanalyzer.config.beanpostprocessor
+
+import com.google.protobuf.GeneratedMessageV3
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+import systems.ajax.malov.stockanalyzer.controller.nats.NatsController
+
+@Component
+class NatsControllerBeanPostProcessor : BeanPostProcessor {
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        if (bean is NatsController<*, *>) {
+            initializeController(bean)
+        }
+        return bean
+    }
+
+    private fun <RequestT : GeneratedMessageV3, ResponseT : GeneratedMessageV3> initializeController(
+        controller: NatsController<RequestT, ResponseT>,
+    ) {
+        // Handle error
+        controller.connection.createDispatcher { message ->
+            Mono.fromSupplier {
+                controller.parser.parseFrom(message.data)
+            }.flatMap {
+                controller.handle(it)
+            }
+                .subscribeOn(Schedulers.boundedElastic())
```

Comment:
why boundedElastic?

Reply:
More threads are better suited for handling and parsing in a dispatcher